src/cudamatrix/cu-kernels.cu

```diff
@@ -2484,6 +2484,42 @@ static void _heaviside(Real*y, const Real*x, MatrixDim d, int src_stride) {
   }
 }
 
+template<typename Real>
+__global__
+static void _exp(Real*y, const Real*x, MatrixDim d, int src_stride) {
+  int i = blockIdx.x * blockDim.x + threadIdx.x;
+  int j = blockIdx.y * blockDim.y + threadIdx.y;
+  int dst_index = i + j * d.stride, src_index = i + j * src_stride;
+  if (i < d.cols && j < d.rows) {
+    Real res = exp(x[src_index]);
+    y[dst_index] = res;
+  }
+}
+
+template<typename Real>
+__global__
+static void _pow(Real*y, const Real*x, Real power, MatrixDim d, int src_stride) {
+  int i = blockIdx.x * blockDim.x + threadIdx.x;
+  int j = blockIdx.y * blockDim.y + threadIdx.y;
+  int dst_index = i + j * d.stride, src_index = i + j * src_stride;
+  if (i < d.cols && j < d.rows) {
+    Real res;
```

Comment:
This function can be substantially simplified (I know this is the same as the previous one).
Just do:
```
y[dst_index] = pow(x[src_index], power);
```
Also please remove the existing _apply_pow kernel and the code that calls it (can just have ApplyPow() be inlined
to call Pow()).
You can do the same for _pow_abs vs. _apply_pow_abs.

Reply:
So when should we make conditions for power? Or is it not necessary any more?

Reply:
not necessary any more.  It's up to the user to ensure it will give reasonable output.